From a99195a702847aa54425f8ae058d49097f7fd4c1 Mon Sep 17 00:00:00 2001
From: Qiang Yu <yuq825@gmail.com>
Date: Mon, 9 Oct 2017 16:04:31 +0800
Subject: [PATCH] drm/lima: change ppmmu dts name

according to
Documentation/devicetree/bindings/gpu/arm,mali-utgard.txt
ppmmu name should be "ppmmuX" instead of "ppXmmu".

Reported-by: Icenowy Zheng <icenowy@aosc.io>
Signed-off-by: Qiang Yu <yuq825@gmail.com>
---
 arch/arm/boot/dts/sun7i-a20.dtsi   | 4 ++--
 arch/arm/boot/dts/sun8i-h3.dtsi    | 4 ++--
 drivers/gpu/drm/lima/lima_device.c | 4 ++--
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/arch/arm/boot/dts/sun7i-a20.dtsi b/arch/arm/boot/dts/sun7i-a20.dtsi
index b4431e7..3edea27 100644
--- a/arch/arm/boot/dts/sun7i-a20.dtsi
+++ b/arch/arm/boot/dts/sun7i-a20.dtsi
@@ -316,8 +316,8 @@
 					   <GIC_SPI 73 IRQ_TYPE_LEVEL_HIGH>,
 					   <GIC_SPI 74 IRQ_TYPE_LEVEL_HIGH>,
 					   <GIC_SPI 75 IRQ_TYPE_LEVEL_HIGH>;
-			interrupt-names = "gp", "gpmmu", "pp0", "pp0mmu", "pmu",
-					  "pp1", "pp1mmu";
+			interrupt-names = "gp", "gpmmu", "pp0", "ppmmu0", "pmu",
+					  "pp1", "ppmmu1";
 			clocks = <&ccu CLK_AHB_GPU>, <&ccu CLK_GPU>;
 			clock-names = "bus", "core";
 			resets = <&ccu RST_GPU>;
diff --git a/arch/arm/boot/dts/sun8i-h3.dtsi b/arch/arm/boot/dts/sun8i-h3.dtsi
index cc2603b..11a6047 100644
--- a/arch/arm/boot/dts/sun8i-h3.dtsi
+++ b/arch/arm/boot/dts/sun8i-h3.dtsi
@@ -91,8 +91,8 @@
 					   <GIC_SPI 101 IRQ_TYPE_LEVEL_HIGH>,
 					   <GIC_SPI 102 IRQ_TYPE_LEVEL_HIGH>,
 					   <GIC_SPI 103 IRQ_TYPE_LEVEL_HIGH>;
-			interrupt-names = "gp", "gpmmu", "pp0", "pp0mmu", "pmu",
-					  "pp1", "pp1mmu";
+			interrupt-names = "gp", "gpmmu", "pp0", "ppmmu0", "pmu",
+					  "pp1", "ppmmu1";
 			clocks = <&ccu CLK_BUS_GPU>, <&ccu CLK_GPU>;
 			clock-names = "bus", "core";
 			resets = <&ccu RST_BUS_GPU>;
diff --git a/drivers/gpu/drm/lima/lima_device.c b/drivers/gpu/drm/lima/lima_device.c
index 6e53d3c..eb6902f 100644
--- a/drivers/gpu/drm/lima/lima_device.c
+++ b/drivers/gpu/drm/lima/lima_device.c
@@ -122,7 +122,7 @@ static int lima_pp_group_init(struct lima_device *dev)
 {
 	int err, i;
 	struct lima_pp *pp;
-	char pp_name[] = "pp0", pp_mmu_name[] = "pp0mmu";
+	char pp_name[] = "pp0", pp_mmu_name[] = "ppmmu0";
 
 	pp = kzalloc(sizeof(*pp), GFP_KERNEL);
 	if (!pp)
@@ -132,7 +132,7 @@ static int lima_pp_group_init(struct lima_device *dev)
 	for (i = 0; i < dev->num_pp; i++) {
 		struct lima_pp_core *core = pp->core + pp->num_core;
 
-		pp_name[2] = '0' + i; pp_mmu_name[2] = '0' + i;
+		pp_name[2] = '0' + i; pp_mmu_name[5] = '0' + i;
 
 		if ((err = lima_init_ip(dev, pp_mmu_name, &core->mmu.ip, LIMA_PPMMU_BASE(i))) ||
 		    (err = lima_mmu_init(&core->mmu))) {
